datajoint/blob.py

```diff
@@ -14,32 +14,54 @@
 from .settings import config
 
 
-mxClassID = dict(
+scalar_type = dict(
     (
         # see http://www.mathworks.com/help/techdoc/apiref/mxclassid.html
-        ("mxUNKNOWN_CLASS", None),
-        ("mxCELL_CLASS", None),
-        ("mxSTRUCT_CLASS", None),
-        ("mxLOGICAL_CLASS", np.dtype("bool")),
-        ("mxCHAR_CLASS", np.dtype("c")),
-        ("mxVOID_CLASS", np.dtype("O")),
-        ("mxDOUBLE_CLASS", np.dtype("float64")),
-        ("mxSINGLE_CLASS", np.dtype("float32")),
-        ("mxINT8_CLASS", np.dtype("int8")),
-        ("mxUINT8_CLASS", np.dtype("uint8")),
-        ("mxINT16_CLASS", np.dtype("int16")),
-        ("mxUINT16_CLASS", np.dtype("uint16")),
-        ("mxINT32_CLASS", np.dtype("int32")),
-        ("mxUINT32_CLASS", np.dtype("uint32")),
-        ("mxINT64_CLASS", np.dtype("int64")),
-        ("mxUINT64_CLASS", np.dtype("uint64")),
-        ("mxFUNCTION_CLASS", None),
+        ("UNKNOWN", None),
+        ("CELL", None),
+        ("STRUCT", None),
+        ("LOGICAL", np.dtype("bool")),
+        ("CHAR", np.dtype("c")),
+        ("VOID", np.dtype("O")),
+        ("DOUBLE", np.dtype("float64")),
+        ("SINGLE", np.dtype("float32")),
+        ("INT8", np.dtype("int8")),
+        ("UINT8", np.dtype("uint8")),
+        ("INT16", np.dtype("int16")),
+        ("UINT16", np.dtype("uint16")),
+        ("INT32", np.dtype("int32")),
+        ("UINT32", np.dtype("uint32")),
+        ("INT64", np.dtype("int64")),
+        ("UINT64", np.dtype("uint64")),
+        ("FUNCTION", None),
+        ("DATETIME64", np.dtype("<M8[us]")),
     )
 )
 
-rev_class_id = {dtype: i for i, dtype in enumerate(mxClassID.values())}
-dtype_list = list(mxClassID.values())
-type_names = list(mxClassID)
+scalar_codes = {
+    np.dtype("bool"): 3,  # LOGICAL
+    np.dtype("c"): 4,  # CHAR
+    np.dtype("O"): 5,  # VOID
+    np.dtype("float64"): 6,  # DOUBLE
+    np.dtype("float32"): 7,  # SINGLE
+    np.dtype("int8"): 8,  # INT8
+    np.dtype("uint8"): 9,  # UINT8
+    np.dtype("int16"): 10,  # INT16
+    np.dtype("uint16"): 11,  # UINT16
+    np.dtype("int32"): 12,  # INT32
+    np.dtype("uint32"): 13,  # UINT32
+    np.dtype("int64"): 14,  # INT64
+    np.dtype("uint64"): 15,  # UINT64
+    np.dtype(
+        "<M8[us]"
+    ): 50,  # Datetime[us], skipped to 50 to accommodate more matlab types
```

Comment:
Let's make it 128, kind of like ascii.  I can envision MATLAB adding 50 datatypes after some time.

Reply:
After review, we decided to put the new types after the 16-bit range since we have up to 32-bits.